paper-button-behavior.html

```diff
@@ -42,7 +60,44 @@
       } else if (this.receivedFocusFromKeyboard) {
         e = 3;
       }
-      this._elevation = e;
+      this.elevation = e;
+    },
+
+    _computeKeyboardClass: function(receivedFocusFromKeyboard) {
+      this.classList.toggle('keyboard-focus', receivedFocusFromKeyboard);
+    },
+
+    _downAction: function(event) {
+      if (!this._ripple) {
+        this._makeRipple(event);
+      }
+    },
+
+    _receivedInitialFocus: function(focused) {
+      if (focused && !this._ripple) {
+        this._makeRipple();
+      }
+    },
+
+    _makeRipple: function(triggeringEvent) {
+      if (!this._ripple) {
+        this._ripple = document.createElement('paper-ripple');
+        Polymer.dom(this.root).appendChild(this._ripple);
+        if (triggeringEvent) {
+          Polymer.dom.flush();
```

Comment:
Would you consider marking up this with some explanatory comments for the future?